cli/medperf/commands/compatibility_test/run.py

```diff
@@ -0,0 +1,243 @@
+        When only the benchmark is provided, it implies the following workflow will be used:
+        - the benchmark's demo dataset is used as the raw data
+        - the benchmark's data preparation cube is used
+        - the benchmark's reference model cube is used
+        - the benchmark's metrics cube is used
+
+        Overriding benchmark's components:
+        - The data prepration, model, and metrics cubes can be overriden by specifying a cube either
+        as an integer (registered) or a path (local). The path can refer either to the mlcube config
+        file or to the mlcube directory containing the mlcube config file.
+        - Instead of using the demo dataset of the benchmark, The input raw data can be overriden by providing:
+            - a demo dataset url and its hash
+            - data path and labels path
+        - A prepared dataset can be directly used. In this case the data preparator cube is never used.
+        The prepared data can be provided by either specifying an integer (registered) or a hash of a
+        locally prepared dataset.
+
+        Whether the benchmark is provided or not, the command will fail either if the user fails to
+        provide a valid complete workflow, or if the user provided extra redundant parameters.
+
+
+        Args:
+            benchmark (int, optional): Benchmark to run the test workflow for
+            data_prep (str, optional): data preparation mlcube uid or local path.
+            model (str, optional): model mlcube uid or local path.
+            evaluator (str, optional): evaluator mlcube uid or local path.
+            data_path (str, optional): path to a local raw data
+            labels_path (str, optional): path to the labels of the local raw data
+            demo_dataset_url (str, optional): Identifier to download the demonstration dataset tarball file.\n
+            See `medperf mlcube submit --help` for more information
+            demo_dataset_hash (str, optional): The hash of the demo dataset tarball file
+            data_uid (str, optional): A prepared dataset UID
+            no_cache (bool): Whether to ignore cached results of the test execution. Defaults to False.
+            offline (bool): Whether to disable communication to the MedPerf server and rely only on
+            local copies of the server assets. Defaults to False.
+
+        Returns:
+            (str): Prepared Dataset UID used for the test. Could be the one provided or a generated one.
+            (dict): Results generated by the test.
+        """
+        logging.info("Starting test execution")
+        test_exec = cls(
+            benchmark,
+            data_prep,
+            model,
+            evaluator,
+            data_path,
+            labels_path,
+            demo_dataset_url,
+            demo_dataset_hash,
+            data_uid,
+            no_cache,
+            offline,
+        )
+        test_exec.validate()
```

Comment:
This is one of the things that caused confusion to me. I couldn't find the `validate` implementation for a while, until I realized it was being inherited. Inspecting the logic of this class requires inspecting the logic of another class to fully understand this behavior